src/DeprecatedCallableInfo.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+/*
+ * This file is part of Twig.
+ *
+ * (c) Fabien Potencier
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Twig;
+
+/**
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+final class DeprecatedCallableInfo
+{
+    private string $type;
+    private string $name;
+
+    public function __construct(
+        private string $package,
+        private string $version,
+        private ?string $altPackage = null,
+        private ?string $altName = null,
+        private ?string $altVersion = null,
+    ) {
+    }
+
+    public function setType(string $type): void
+    {
+        $this->type = $type;
+    }
+
+    public function setName(string $name): void
+    {
+        $this->name = $name;
+    }
+
+    public function triggerDeprecation(?string $file = null, ?int $line = null): void
+    {
+        $message = \sprintf('Twig %s "%s" is deprecated', ucfirst($this->type), $this->name);
+
+        if ($this->altName) {
+            $message .= \sprintf('. Use "%s"', $this->altName);
+            if ($this->altPackage) {
+                $message .= \sprintf(' from the "%s" package', $this->altPackage);
+            }
+            if ($this->altVersion) {
+                $message .= \sprintf(' (available since "%s")', $this->altVersion);
+            }
+            $message .= ' instead';
+        }
+
+        if ($file) {
+            $message .= \sprintf(' in %s at line %d.', $file, $line);
```

Comment:
What if `$line` is `null` when providing `$file` ?

Reply:
Fixed, test added